src/test/volume.test.ts

```diff
@@ -0,0 +1,129 @@
+import { expect } from "chai";
+import { Vector2, Vector3 } from "three";
+
+import Volume, { ImageInfo } from "../Volume";
+import VolumeMaker from "../VolumeMaker";
+import { LUT_ARRAY_LENGTH } from "../Histogram";
+import Channel from "../Channel";
+
+// PREPARE SOME TEST DATA TO TRY TO DISPLAY A VOLUME.
+const testimgdata: ImageInfo = {
+  name: "AICS-10_5_5",
+  version: "0.0.0",
+
+  originalSize: new Vector2(306, 494),
+  atlasDims: new Vector2(7, 10),
+  volumeSize: new Vector3(204, 292, 65),
+  regionSize: new Vector3(204, 292, 65),
+  regionOffset: new Vector3(0, 0, 0),
+  pixelSize: new Vector3(0.065, 0.065, 0.29),
+  spatialUnit: "",
+
+  numChannels: 9,
+  channelNames: [
+    "DRAQ5",
+    "EGFP",
+    "Hoechst 33258",
+    "TL Brightfield",
+    "SEG_STRUCT",
+    "SEG_Memb",
+    "SEG_DNA",
+    "CON_Memb",
+    "CON_DNA",
+  ],
+
+  times: 1,
+  timeScale: 1,
+  timeUnit: "",
+
+  transform: {
+    translation: new Vector3(0, 0, 0),
+    rotation: new Vector3(0, 0, 0),
+  },
+};
+
+function checkVolumeConstruction(v: Volume, imgdata: ImageInfo) {
+  expect(v).to.be.a("Object");
+  expect(v.isLoaded()).to.not.be.ok;
+
+  const { originalSize, volumeSize, pixelSize } = imgdata;
+  const physicalSize = new Vector3(originalSize.x, originalSize.y, volumeSize.z).multiply(pixelSize);
+  expect(v.physicalSize.x).to.equal(physicalSize.x);
+  expect(v.physicalSize.y).to.equal(physicalSize.y);
+  expect(v.physicalSize.z).to.equal(physicalSize.z);
+  expect(v.channelNames.length).to.equal(imgdata.numChannels);
+  expect(v.channels.length).to.equal(imgdata.numChannels);
+
+  const mx = Math.max(Math.max(v.normPhysicalSize.x, v.normPhysicalSize.y), v.normPhysicalSize.z);
+  expect(mx).to.equal(1.0);
+
+  // TODO cleanup?
+  // expect(v.x).to.equal(imgdata.tile_width);
+  // expect(v.y).to.equal(imgdata.tile_height);
+  // expect(v.z).to.equal(imgdata.tiles);
```

Comment:
What's the reasoning for keeping these as comments? Is there an aspect of deleting this that's currently uncertain?

Reply:
Thought I'd done a thorough enough pass over this PR to clean out all the things I'd tentatively commented out. Apparently not. I'll give it another look over

Reply:
Done! I'll just comment here and resolve comments below which this pass mopped up